### PR TITLE
Correct names of pushforward functions in `VectorFemField`

### PR DESCRIPTION
**Pull request by anlavandier**
Rename methods of `VectorFemField` class:

- `pushforward` -> `pushforward_field`
- `pushforward_grid`-> `pushforward_fields`
- `pushforward_regular_tensor_grid` -> `pushforward_fields_regular_tensor_grid`

Also apply some small corrections to the docstrings.

### CODE REVIEW

File: psydac/fem/vector.py [hunk 1]
```diff
@@ -225,11 +224,11 @@ def pushforward_grid(self, grid, *fields, mapping=None, npts_per_cell=None):
...
                 grid[i] = np.reshape(grid[i], newshape=(ncells_i, npts_per_cell[i]))
 
-            pushed_fields = self.pushforward_regular_tensor_grid(grid, *fields, mapping=mapping)
+            pushed_fields = self.pushforward_fields_regular_tensor_grid(grid, *fields, mapping=mapping)
             # return a list of list of C-contiguous arrays, one list for each field
             # with one array for each dimension.
-            return [[np.ascontiguousarray(pushed_fields[..., j, i]) for j in range(self._ldim)]
-                    for i in range(len(fields))]
+            return [[np.ascontiguousarray(pushed_fields[..., i, j]) for i in range(self._ldim)]
+                    for j in range(len(fields))]
```
Comment (yguclu) [inline psydac/fem/vector.py:231]: Can you please leave the indexing as before, and simply swap the two ranges? I think that's clearer

File: psydac/fem/vector.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def integral( self, f ):
         raise NotImplementedError( "VectorFemSpace not yet operational" )
 
     # ...
-    def pushforward_grid(self, grid, *fields, mapping=None, npts_per_cell=None):
+    def pushforward_fields(self, grid, *fields, mapping=None, npts_per_cell=None):
```
Comment (yguclu) [inline psydac/fem/vector.py:166]: If you want `mapping` to be a required keyword-only argument, you can do so by adding the `*` separator before it, and removing the default value `None`:
```
def pushforward_fields(self, grid, *fields, *, mapping, npts_per_cell=None):
```

**Review by yguclu — CHANGES_REQUESTED** [review]

File: psydac/fem/vector.py [hunk 3]
```diff
@@ -268,11 +267,12 @@ def pushforward_regular_tensor_grid(self, grid, *fields, mapping=None, parent_ki
...
             evaluation points in the same direction.
 
-        fields: List of psydac.fem.basic.FemField
+        fields: tuple of psydac.fem.basic.FemField
+            List of fields to evaluate.
 
         mapping: psydac.mapping.SplineMapping
             Mapping on which to push-forward
-
+
```
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/fem/vector.py]: Done d58dc62

File: psydac/fem/vector.py (hunk 1 above)
Comment (anlavandier) [reply psydac/fem/vector.py]: Done d58dc62

File: psydac/fem/vector.py (hunk 2 above)
Comment (anlavandier) [reply psydac/fem/vector.py:166]: I checked and I can't do it because I have a starred argument `*fields` beforehand.

Comment (anlavandier) [reply psydac/fem/vector.py:166]: I checked again and I can do it without adding the `*` because of the starred argument.

Comment (anlavandier) [reply psydac/fem/vector.py]: Done in [b214861](https://github.com/pyccel/psydac/pull/212/commits/b214861020dd5759840cfd012919c8452089a99d)

**Review by yguclu — APPROVED** [review]

**Pull request merged**